src/cocotb/_scheduler.py

```diff
@@ -201,17 +201,19 @@ def __init__(self) -> None:
         # indexed by trigger
         self._trigger2tasks: Dict[Trigger, list[Task]] = insertion_ordered_dict()
 
-        self._scheduled_tasks: OrderedDict[Task[Any], Union[BaseException, None]] = (
+        self._scheduled_tasks: OrderedDict[Task[Any], Optional[BaseException]] = (
             OrderedDict()
         )
-        self._pending_threads = []
-        self._pending_events = []  # Events we need to call set on once we've unwound
+        self._pending_threads: list[external_waiter] = []
+        self._pending_events: list[
+            Event
+        ] = []  # Events we need to call set on once we've unwound
```

Comment:
You can delete the comment, it's not useful, and the code will be a bit prettier.